plugins/flytekit-kf-pytorch/flytekitplugins/kfpytorch/pod_template.py

```diff
@@ -0,0 +1,36 @@
+from kubernetes.client import V1Container, V1EmptyDirVolumeSource, V1PodSpec, V1Volume, V1VolumeMount
+
+from flytekit.core.pod_template import PodTemplate
+
+
+def add_shared_mem_volume_to_pod_template(pod_template: PodTemplate) -> None:
+    """Add shared memory volume and volume mount to the pod template."""
+    shm_volume = V1Volume(name="shm", empty_dir=V1EmptyDirVolumeSource(medium="Memory"))
+    shm_volume_mount = V1VolumeMount(name="shm", mount_path="/dev/shm")
+
+    if pod_template.pod_spec is None:
+        pod_template.pod_spec = V1PodSpec()
+
+    if pod_template.pod_spec.containers is None:
+        pod_template.pod_spec.containers = []
+
+    if pod_template.pod_spec.volumes is None:
+        pod_template.pod_spec.volumes = []
+    pod_template.pod_spec.volumes.append(shm_volume)
+
+    num_containers = len(pod_template.pod_spec.containers)
```

Comment:
Nit: Can we do the error checking here? 

```python
if num_container >= 2:
    raise ValueError(...)

if num_containers != 1:
    pod_template.pod_spec.containers.append(V1Container(name="primary"))
```

Reply:
https://github.com/flyteorg/flytekit/pull/2543/commits/d03b6af5f2a7d195ab4191e6ed07a0e47b1d1aa9